src/core/Ros.js

```diff
@@ -48,7 +50,16 @@ Ros.prototype.__proto__ = EventEmitter2.prototype;
  * @param url - WebSocket URL for Rosbridge
  */
 Ros.prototype.connect = function(url) {
-  this.socket = assign(new WebSocket(url), socketAdapter(this));
+  if (this.transportLibrary === 'socket.io') {
+    this.socket = assign(io(url, {'force new connection': true}), socketAdapter(this));
+    this.socket.on('connect', this.socket.onopen);
+    this.socket.on('data', this.socket.onmessage);
+    this.socket.on('close', this.socket.onclose);
+    this.socket.on('error', this.socket.onerror);
```

Comment:
Would it be better to modify src/core to accept some kind of transport object instead? This would provide a common abstraction to prevent changes in src/core from breaking other transports. It then additionally does not require hard coding every possible transport in src/core. I've been looking into this as I am looking to implement rosbridge over WebRTC.